pkg/kubecost/totals.go

```diff
@@ -214,17 +268,34 @@ func ComputeAssetTotals(as *AssetSet, prop AssetProperty) map[string]*AssetTotal
 				adjustmentRate = node.TotalCost() / (node.TotalCost() - node.Adjustment())
 			}
 
-			totalCPUCost := node.CPUCost * (1.0 - node.Discount)
-			cpuCost := totalCPUCost * adjustmentRate
-			cpuCostAdjustment := totalCPUCost - cpuCost
+			// 1. Start with raw, measured resource cost
+			// 2. Apply discount to get discounted resource cost
+			// 3. Apply adjustment to get final "adjusted" resource cost
+			// 4. Subtract (3 - 2) to get adjustment in doller-terms
+			// 5. Use (2 + 4) as total cost, so (2) is "cost" and (4) is "adjustment"
 
-			totalGPUCost := node.GPUCost * (1.0 - node.Discount)
-			gpuCost := totalGPUCost * adjustmentRate
-			gpuCostAdjustment := totalGPUCost - gpuCost
+			// Example:
+			// - node.CPUCost   = 10.00
+			// - node.Discount  =  0.20  // We assume a 20% discount
+			// - adjustmentRate =  0.75  // CUR says we need to reduce to 75% of our post-discount node cost
```

Comment:
@nikovacevic This comment is superb. For my own education, I didn't realize that billing worked like this. Is node.Discount something like an RI discount and adjustmentRate is like an account- or project-wide discount negotiated with AWS?

Reply:
`node.Discount` is a combination of settings and special knowledge, encoded in:
```
cm.Provider.CombinedDiscountForNode(node.NodeType, node.Preemptible, discount, negotiatedDiscount)
```

`adjustmentRate` is just a way of spreading the node's single cost `adjustment` across it's constituent resources.

So, on GKE your discount might be an expected 30% for an n1 with sustained use, or 20% for an n2 sustained use, or 0% for an e2 sustained use. Then, after we reconcile with the CUR, a single adjustment is added to or subtracted from the node's cost, but we don't know how that applies to CPU, RAM and GPU costs. So we use adjustment rate to spread the adjustment across the resources. With discount _and_ adjustment applied in the correct order, we get a node with CPU, RAM, and GPU that sum to agree with the CUR.